### PR TITLE
improvement: switch from uglify to terser webpack plugin

### PR DESCRIPTION
**Pull request by EddyVinck**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Changelog have been added / updated (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Changes UglifyJsPlugin to TerserWebpackPlugin https://github.com/deity-io/falcon/issues/601

**What is the current behavior? (You can also link to an open issue here)**
The bahavior is the same

**What is the new behavior (if this is a feature change)?**
The new behavior is the same, except Terser is maintained

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**
I tested it by running `yarn start` in the root directory, but I would like some help to verify that nothing is broken.

**Other information:**
Terser is a fork of Uglify, like Daniel said. I checked and the API should be compatible. All the options used in Falcon are supported in the same configuration structure by Terser.

### CODE REVIEW

File: packages/falcon-client/package.json [hunk 1]
```diff
@@ -130,7 +130,8 @@
     "react-dom": "16.10.2",
     "react-router-dom": "5.0.1",
     "react-test-renderer": "16.10.2",
-    "supertest": "4.0.2"
+    "supertest": "4.0.2",
+    "terser-webpack-plugin": "^2.2.1"
```
Comment (tutok) [inline packages/falcon-client/package.json]: please remove `uglifyjs-webpack-plugin`

File: packages/falcon-client/build-utils/webpack/config/create.js [hunk 2]
```diff
@@ -487,8 +487,8 @@ module.exports = (target = 'web', options) => {
         ...config.optimization,
         minimize: true,
         minimizer: [
-          new UglifyJsPlugin({
-            uglifyOptions: {
+          new TerserPlugin({
+            terserOptions: {
```
Comment (tutok) [inline packages/falcon-client/build-utils/webpack/config/create.js]: looks like option `terserOptions.parallel` is set by default into `true`, so maybe we can remove it (with its comment too)

---
I also did a review of other settings and it seems that they need to stay as they were, `create-react-app` uses similar settings.

**Comment by tutok** [comment]
`yarn build:prod` produces `5KB` larger bundle, so no improvements 😞

**Comment by EddyVinck** [comment]
> `yarn build:prod` produces `5KB` larger bundle, so no improvements

larger than when using uglify? t's for the better I suppose but that's kind of sad :(

File: packages/falcon-client/package.json (hunk 1 above)
Comment (tutok) [reply packages/falcon-client/package.json]: I think you should also run `yarn install` to update `yarn.lock`.
If you will not do this right now, someone could fetch your changes latter (no matter if from this branch or latter from `dev`) and after `yarn install`, he will have local changes.

**Comment by tutok** [comment]
@EddyVinck  can you please resolve conflicts? after that I think we can merge

**Comment by EddyVinck** [comment]
@tutok this should be good to go now

**Pull request merged**